python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -0,0 +1,251 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import time
+import tensorflow as tf
+from typing import Dict, Optional, List
+from bigdl.nano.utils.inference.common.base_optimizer import BaseInferenceOptimizer
+from bigdl.nano.utils.inference.common.checker import available_acceleration_combination
+from bigdl.nano.utils.inference.common.utils import AccelerationOption,\
+    throughput_calculate_helper, format_optimize_result
+from bigdl.nano.tf.keras import Model as NanoModel
+from bigdl.nano.utils.log4Error import invalidInputError
+from tensorflow.keras import Model as Model
+from tensorflow.data import Dataset
+from tensorflow.keras.metrics import Metric
+
+
+class InferenceOptimizer(BaseInferenceOptimizer):
+
+    # acceleration method combinations, developers may want to register some new
+    # combinations here
+    ALL_INFERENCE_ACCELERATION_METHOD: Dict = \
+        {  # type: ignore
+            "original": AccelerationOption(),
+            "int8": AccelerationOption(inc=True),
+            "openvino_fp32": AccelerationOption(openvino=True),
```

Comment:
Is it possible to refactor the optimization logic to a class similar to this `AccelerationOption` so that it is clear which options does what e.g.
```python
class Acceleration:
    def __init__(inc:bool, openvion:bool, ...):
        pass
    
    def optimize(model, **other_kwargs):
        # optimization logic
        return optimized_model
```

Reply:
I think it is not urgent though.

Reply:
Thanks, will consider this after basic logic is ready and try some refactor.